docs/guides/advanced-routing.md

```diff
@@ -0,0 +1,332 @@
+# Advanced HTTP Routing
+
+In this guide we will be configuring a more advanced set of routing rules with multiple applications. These rules
+will showcase different paths, parameters, and methods used to fine-tune the flow of traffic. For an introduction
+to exposing your application, it is recommended to go through the
+[basic guide](/docs/guides/how-to-direct-traffic-to-your-app.md) first.
+
+The following image shows the traffic flow that we will be creating with these rules.
+
+![Traffic Flow Diagram](/docs/images/advanced-routing.png)
+
+The goal is to create a set of rules that will result in users' requests being sent to specific backends based on
+the conditions in which the request is sent.
```

Comment:
I know the diagram is pretty self-explanatory to us, but could we briefly discuss the goals of this architecture before implementation? I want to make sure the reader gets a really clear idea of what our architecture is intended to accomplish before implementation. Something along the lines of:
- We have two versions of the coffee service. v1 traffic needs to be directed to the old application, while v2 traffic needs to be directed towards the new implementation.
- The GET and POST operations on the Tea API are hosted by two separate applications. We need to ensure all GETs go to the pod Tea, and all POSTs go to the pod tea-post.
- Both the Tea and Coffee APIs share the same gateway.